src/main/scala/firrtl/package.scala

```diff
@@ -8,5 +8,6 @@ package object firrtl {
 
   /* Options as annotations compatibility items */
   @deprecated("Use firrtl.stage.TargetDirAnnotation", "3.2")
+  type TargetDirAnnotation = firrtl.stage.TargetDirAnnotation
   val TargetDirAnnotation = firrtl.stage.TargetDirAnnotation
```

Comment:
I think you need to deprecate both? I'm not sure though

Reply:
Good catch. One odd interaction that I did not expect is that the deprecation warnings will only trigger when doing an assignment. I expected them to trigger when matching on type or unapplying, but it doesn't.
```scala
/* Both of these trigger compile time deprecation warnings */
val x = TargetDirAnnotation()
val y = TargetDirAnnotation

/* Neither of these match statements will result in deprecation warnings */
x match {
  case TargetDirAnnotation(xx) => println(xx)
  case xx: TargetDirAnnotation => println("hello")
}
```

Updated in the amended commit.